CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+# Minimum CMake required
+cmake_minimum_required(VERSION 3.1)
+
+# Project
+project(onnxir C CXX)
+
+# Set C++11 as standard for the whole project
+set(CMAKE_CXX_STANDARD 11)
+
+set(ONNXIR_ROOT ${PROJECT_SOURCE_DIR})
+if (WIN32)
+    # parallel build
+    add_compile_options(/MP)
+    add_compile_options(/WX)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WX")
+    set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} /WX")
+endif()
+add_subdirectory(third_party/pybind11)
+FIND_PACKAGE(Protobuf REQUIRED)
+include_directories(
+    ${ONNXIR_ROOT}
+    ${CMAKE_CURRENT_BINARY_DIR}
+    # External dependencies.
+    ${PROTOBUF_INCLUDE_DIRS}
+    ${PYBIND11_INCLUDE_DIR}
+    ${PYTHON_INCLUDE_DIRS}
+)
+
+# function(RELATIVE_PROTOBUF_GENERATE_CPP SRCS HDRS ROOT_DIR)
+# from https://github.com/tensorflow/tensorflow/blob/d2c3b873c6f8ff999a2e4ee707a84ff00d9c15a5/tensorflow/contrib/cmake/tf_core_framework.cmake
+# to solve the problem that customized dir can't be specified when calling PROTOBUF_GENERATE_CPP.
+function(RELATIVE_PROTOBUF_GENERATE_CPP SRCS HDRS ROOT_DIR)
+  if(NOT ARGN)
+    message(SEND_ERROR "Error: RELATIVE_PROTOBUF_GENERATE_CPP() called without any proto files")
+    return()
+  endif()
+
+  set(${SRCS})
+  set(${HDRS})
+  foreach(FIL ${ARGN})
+    message(${FIL})
+    set(ABS_FIL ${ROOT_DIR}/${FIL})
+    get_filename_component(FIL_WE ${FIL} NAME_WE)
+    get_filename_component(FIL_DIR ${ABS_FIL} DIRECTORY)
+    file(RELATIVE_PATH REL_DIR ${ROOT_DIR} ${FIL_DIR})
+
+    list(APPEND ${SRCS} "${CMAKE_CURRENT_BINARY_DIR}/${REL_DIR}/${FIL_WE}.pb.cc")
+    list(APPEND ${HDRS} "${CMAKE_CURRENT_BINARY_DIR}/${REL_DIR}/${FIL_WE}.pb.h")
+
+    add_custom_command(
```

Comment:
Note that if you want to build dlls, you will want to handle that with dllexport symbols. This is how we do it in c2:

https://github.com/caffe2/caffe2/blob/b7d983f255ef5496474f1ea188edb5e0ac442761/cmake/ProtoBuf.cmake#L165-L184

Not sure if I pointed that out before, so I'll just put it up here again just in case.

Reply:
thank you! Will add it.